Frontend/app/src/main/java/io/happybugs/happybugs/activity/ReportActivity.java

```diff
@@ -0,0 +1,191 @@
+
+        // Create question buttons and 'Save and Exit' button.
+        buttons = new ReportButtons((Button) findViewById(R.id.whatBtn),
+                (Button) findViewById(R.id.whereBtn), (Button) findViewById(R.id.whenBtn),
+                (Button) findViewById(R.id.whoBtn), (Button) findViewById(R.id.detailsBtn),
+                (Button) findViewById(R.id.saveBtn));
+
+        // Create answer texts.
+        editTexts = new ReportEditTexts((EditText) findViewById(R.id.whatText),
+                (EditText) findViewById(R.id.whereText), (EditText) findViewById(R.id.whenText),
+                (EditText) findViewById(R.id.whoText), (EditText) findViewById(R.id.detailsText));
+
+        // Create checkboxes.
+        // Checkboxes are enabled whenever a question is answered.
+        checkBoxes = new ReportCheckBoxes((CheckBox) findViewById(R.id.whatCheck),
+                (CheckBox) findViewById(R.id.whereCheck), (CheckBox) findViewById(R.id.whenCheck),
+                (CheckBox) findViewById(R.id.whoCheck), (CheckBox) findViewById(R.id.detailsCheck));
+
+        // Create underlines separating questions.
+        views = new ReportViews((View) findViewById(R.id.whatView), (View) findViewById(R.id.whereView),
+                (View) findViewById(R.id.whenView), (View) findViewById(R.id.whoView),
+                (View) findViewById(R.id.detailsView));
+
+        buttons.whatBtn.setOnClickListener(this);
+        buttons.whereBtn.setOnClickListener(this);
+        buttons.whenBtn.setOnClickListener(this);
+        buttons.whoBtn.setOnClickListener(this);
+        buttons.detailsBtn.setOnClickListener(this);
+        buttons.saveBtn.setOnClickListener(this);
+    }
+
+    @Override
+    public void onClick(View v) {
+        switch (v.getId()){
+            case R.id.whatBtn:
+                questionClickEvent(editTexts, editTexts.whatText, checkBoxes.whatCheck,
+                        buttons.saveBtn, views, views.whatView);
+                break;
+            case R.id.whereBtn:
+                questionClickEvent(editTexts, editTexts.whereText, checkBoxes.whereCheck,
+                        buttons.saveBtn, views, views.whereView);
+                break;
+            case R.id.whenBtn:
+                questionClickEvent(editTexts, editTexts.whenText, checkBoxes.whenCheck,
+                        buttons.saveBtn, views, views.whenView);
+                break;
+            case R.id.whoBtn:
+                questionClickEvent(editTexts, editTexts.whoText, checkBoxes.whoCheck,
+                        buttons.saveBtn, views, views.whoView);
+                break;
+            case R.id.detailsBtn:
+                questionClickEvent(editTexts, editTexts.detailsText, checkBoxes.detailsCheck,
+                        buttons.saveBtn, views, views.detailsView);
+                break;
+            case R.id.saveBtn:
+                sendReportData();
+                break;
+        }
```

Comment:
Case when R.id.saveBtn is clicked is when the new created report's data is sent to server. Other cases for actions needed when a question is clicked(e.g. open and collapse answer boxes..). If a default case is needed, I think it would only be for the R.id.saveBtn case. Errors for POST data will be handled further. Do you still think a default case would be needed to fall back on?